crates/subspace-node/src/commands/domain_key.rs

```diff
@@ -0,0 +1,128 @@
+use crate::commands::shared::{
+    derive_keypair, init_logger, store_key_in_keystore, KeystoreOptions,
+};
+use bip39::Mnemonic;
+use clap::Parser;
+use sc_cli::{Error, KeystoreParams};
+use sc_service::config::KeystoreConfig;
+use sp_core::crypto::{ExposeSecret, SecretString};
+use sp_core::Pair;
+use sp_domains::DomainId;
+use std::path::PathBuf;
+use tracing::{info, warn};
+
+/// Options for creating domain key
+#[derive(Debug, Parser)]
+pub struct CreateDomainKeyOptions {
+    /// Base path where to store node files
+    #[arg(long)]
+    base_path: PathBuf,
+    /// ID of the domain to store key for
+    #[arg(long, required = true)]
+    domain_id: DomainId,
+    /// Options for domain keystore
+    #[clap(flatten)]
+    keystore_options: KeystoreOptions,
+}
+
+pub fn create_domain_key(options: CreateDomainKeyOptions) -> Result<(), Error> {
+    init_logger();
+
+    let CreateDomainKeyOptions {
+        base_path,
+        domain_id,
+        keystore_options,
+    } = options;
+    let domain_path = base_path.join("domains").join(domain_id.to_string());
+
+    let keystore_params = KeystoreParams {
+        keystore_path: None,
+        password_interactive: keystore_options.keystore_password_interactive,
+        password: keystore_options.keystore_password,
+        password_filename: keystore_options.keystore_password_filename,
+    };
+
+    let keystore_config = keystore_params.keystore_config(&domain_path)?;
+
+    let (path, password) = match &keystore_config {
+        KeystoreConfig::Path { path, password, .. } => (path.clone(), password.clone()),
+        KeystoreConfig::InMemory => {
+            unreachable!("Just constructed non-memory keystore config; qed");
```

Comment:
```suggestion
            unreachable!("Just construct non-memory keystore config; qed");
```

Reply:
Why?

Reply:
I'm not sure I understand the comment then. Mostly confused about what that statement actually means. I'm assuming you want the user to pass suri ?

Reply:
No, this is a match on `KeystoreConfig` and `keystore_params.keystore_config()` always constructs `Keystore::Path` variant, it never does memory keystore.